packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -238,45 +238,6 @@ class PipelineController extends CanvasController {
     }
   }
 
-  setInvalidNode(pipelineID: string, nodeID: string) {
-    const node = this.getNode(nodeID, pipelineID);
-    if (node.type !== "execution_node") {
-      return;
-    }
-    // TODO: this shouldn't be "red"
-    const image =
-      "data:image/svg+xml;utf8," +
-      encodeURIComponent(`<svg
-            xmlns="http://www.w3.org/2000/svg"
-            width="100"
-            viewBox="0 0 22 22"
-          >
-            <text
-              x="11"
-              y="16.5"
-              text-anchor="middle"
-              fill="red" 
-              font-family="'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif"
-              font-size="15px"
-            >
-              ?
-            </text>
-          </svg>`);
-    this.setNodeProperties(
-      node.id,
-      {
-        app_data: {
-          ...node.app_data,
-          invalidNodeError: `"${node.op}" is an unsupported node type`,
-        },
-        description: undefined,
-        image: image,
-      },
-      pipelineID
-    );
-    this.setNodeLabel(node.id, "unsupported node", pipelineID);
-  }
-
```

Comment:
was this just an unused function that was meant to do this validation?

Reply:
This was originally intended to catch nodes that were only available to kfp/airflow when switching between pipeline type. Since it’s not possible to switch pipeline types this turned into dead code. This piece of code turns the icon for the node into a question mark and changes the label to “invalid node” which is poor user experience, because you can’t tell what the node was intended to be